### ISSUE TITLE
Regression with druid 28 - round generates java.lang.RuntimeException when applied on sys tables

### ISSUE THREAD

**Issue by ochautard** [apache/druid#15563]
Please provide a detailed title (e.g. "Broker crashes when using TopN query with Bound filter" instead of just "Broker crashes").

### Affected Version

Working on druid 27; but not working on druid 28

### Description

I do not think that cluster size or configuration has any impact.

I have some basic query to check the size of druid segments from the druid console:
```SQL
SELECT datasource, avg(size)/1024/1024 as size_mb FROM sys.segments group by datasource
```
With druid 27.0.0, the "size_mb" was automatically converted to a bigint.

With druid 28.0.0, the "size_mb" is no more converted and is returned as a double. Fair enough.

But when I tried to apply some rounding on it, it does not work:
```SQL
round(avg(size)/1024/1024)
round(cast(avg(size)/1024/1024 as DOUBLE))
round(cast(avg(3000)/1024/1024 as DOUBLE),1)
```

In any case, in the druid console, I get the following error:
```
Error: RUNTIME_FAILURE (OPERATOR)
cannot translate call ROUND($t7)
java.lang.RuntimeException
```

And I also have this warning in the logs of the broker (but no stacktrace):
```
2023-12-12T12:58:11,484 WARN [sql[bb34e9cd-7422-4c38-93af-2ee86d850e50]] org.apache.druid.sql.http.SqlResource - Exception while processing sqlQueryId[bb34e9cd-7422-4c38-93af-2ee86d850e50] (org.apache.druid.error.DruidException: cannot translate call ROUND($t18))
```

Note: I do NOT get any issue  when querying directly on a value (for instance `round(0.4134190559387207 , 1)`) or when querying on a datasource (it only happens on sys tables.

**Comment by aleksi75** [comment]
I also tried the above example: 
SELECT datasource, round(avg(size)/1024/1024) as size_mb FROM sys.segments group by datasource

Druid 27: works
Druid 28: Error: RUNTIME_FAILURE (OPERATOR), cannot translate call ROUND($t7)

Maybe there is a 'connection' with the issue I created:
https://github.com/apache/druid/issues/15562

**Comment by cryptoe** [comment]
Issue fixed with #15562 . 
Thanks for reporting it. 
The fix would be part of druid 29 release.

**Comment by hevansDev** [comment]
Observed the same issue with SUBSTRING in Druid 29.0.1, was there ever a fix for this?

```
SELECT
SUBSTRING("segment_id",5) AS "segment_s", "num_rows"
FROM "sys"."segments" WHERE "datasource"='adsb-json' LIMIT 7
```
```
Error: RUNTIME_FAILURE (OPERATOR)

cannot translate call SUBSTRING($t0, $t19)

java.lang.RuntimeException
```
